### PR TITLE
GH-45071: [Packaging][Docs] Fix NumPy v2 include directory for Emscripten, and update Pyodide-related documentation

### LINKED ISSUE #45071: [Packaging][Docs] Pyodide builds have incorrect NumPy 2.0 header location and assume `pyodide-build` to be the same version as Pyodide
### Describe the bug, including details regarding any error messages, version, and platform.

Hi there, we're trying to build PyArrow against NumPy v2 as a part of pyodide/pyodide#4925 and we are targeting an imminent 0.27 release soon after. I've opened this issue to put to light, that:
- the Pyodide builds are against NumPy 1.X right now, but building against 2.X has a problem because the NumPy headers were moved from `numpy/core/include` to `numpy/_core/include/`:

	https://github.com/apache/arrow/blob/b655852b260d3b8c3fe457795df0f42a2ff9c98c/python/CMakeLists.txt#L163-L177

- the documentation mentions in a few places that the version of `pyodide-build` must match the Pyodide version – this is no longer the case, since, we have [unvendored `pyodide-build` to a separate repository](https://github.com/pyodide/pyodide-build) and now [maintain a list of compatible versions against Pyodide](https://raw.githubusercontent.com/pyodide/pyodide/main/pyodide-cross-build-environments.json).
- the Pyodide and Emscripten versions are slightly out of date in a few places

I already have a pull request coming for addressing these changes, but since this was also concerning CI/packaging and not just documentation, the Contributing guide suggested that I should open an issue first before opening a pull request.

Thanks for your time!

### Component(s)

Python

### PR DESCRIPTION
**Pull request by agriyakhetarpal**
### Rationale for this change

This change would allow building PyArrow correctly with NumPy 1.X and NumPy 2.X, since we are trying to do the latter for pyodide/pyodide#4925. This PR closes gh-45071.

### What changes are included in this PR?

This PR
- issues a correction for the NumPy header files when building under Emscripten
- updates Pyodide-related build instructions

### Are these changes tested?

Yes, working here: https://github.com/agriyakhetarpal/pyodide-numpy-2.0-rebuilds/actions/runs/12399351376/job/34619554658#step:8:4547 via https://github.com/agriyakhetarpal/pyodide/commit/b651698ab31b9fe277437543dd483eb19aaa2462 that applies a subset of the changes as a patch (the CI job is failing for unrelated reasons, please ignore).

### Are there any user-facing changes?

Yes, users trying to build a WASM wheel via Pyodide are now requested to use newer Pyodide and Emscripten versions, and the latest stable version of `pyodide-build` available.

* GitHub Issue: #45071

### CODE REVIEW

File: python/CMakeLists.txt [hunk 1]
```diff
@@ -166,8 +166,17 @@ if($ENV{PYODIDE})
...
   set(Python3_INCLUDE_DIR $ENV{PYTHONINCLUDE})
   set(Python3_LIBRARY $ENV{CPYTHONLIB})
-  set(Python3_NumPy_INCLUDE_DIR $ENV{NUMPY_LIB}/core/include)
   set(Python3_EXECUTABLE)
+  execute_process(COMMAND ${Python3_EXECUTABLE} -c
+                          "import numpy; print(numpy.__version__)"
+                  OUTPUT_VARIABLE PYODIDE_NUMPY_VERSION
+                  OUTPUT_STRIP_TRAILING_WHITESPACE)
+  string(REGEX MATCH "^([0-9]+)" PYODIDE_NUMPY_MAJOR_VERSION ${PYODIDE_NUMPY_VERSION})
+  if(NUMPY_MAJOR_VERSION GREATER_EQUAL 2)
```
Comment (agriyakhetarpal) [inline python/CMakeLists.txt:175]:
```suggestion
  if(PYODIDE_NUMPY_MAJOR_VERSION GREATER_EQUAL 2)
```

**Comment by kou** [comment]
@github-actions crossbow submit -g python

**Comment by agriyakhetarpal** [comment]
Thanks for triggering CI! Looks like the failure is unrelated, it's just a timeout.

**Comment by raulcd** [comment]
> Thanks for triggering CI! Looks like the failure is unrelated, it's just a timeout.

Yes, it seems a timeout issue but it's pretty consistent, I just triggered it a 4th time, and it has never failed on our nightly CI. The other CI failures are unrelated.

**Comment by agriyakhetarpal** [comment]
Glad that it passed :D

I would recommend switching to using the Pyodide virtual environment `pyodide venv` for running the test suite, by the way – while `python/scripts/run_emscripten_tests.py` tests on Chrome and Firefox as well, the Node.js tests in it could be swapped out for it, and we found to be more reliable in projects such as SymPy.

**Review by kou — APPROVED** [review]
+1

I did more tries and they are stable now. (Why...?)

Could you open new issues for these improvement ideas?

**Comment by agriyakhetarpal** [comment]
I have no idea either, though FWIW, timeouts and flakes with `selenium` in our own test suite are pretty common.

I'll open a new issue about it, sure!

**Pull request merged**